_posts/2020/2020-12-02-load-testing-as-development-tool.md

```diff
@@ -0,0 +1,177 @@
+It can be found here: [Artillery Github Repo](https://github.com/artilleryio/artillery){:target="_blank"}
+
+It is available as a NPM (Node Package Manager) package, so you must either have it installed or run it from inside a docker container (believe me, this is possible, and can save time and space in your disk).
+The *npm* command to install it is as follows: `$ npm install -g artillery`
+
+Differently from the previous tool, Artillery requires a `yaml` file that describes the load test scenario.
+It then reads this file and executes the steps described.
+A couple advantages of using a file for running load tests:
+* Descriptive: easy to read and understand what will be executed
+* Can be version controlled
+
+Here is an example of such a yaml file:
+
+```yaml
+config:
+  target: 'http://httpbin.org'
+  phases:
+    - duration: 20 # How many seconds this test will run for
+      arrivalRate: 20 # The number of requests per second
+      name: "The real deal"
+  defaults:
+    headers:
+      Authorization: 'Bearer <insert the token here>'
+scenarios:
+  - flow:
+    - get:
+        url: "/get"
+```
+
+This yaml file is very basic and is a rough equivalent to the command we used with vegeta earlier.
+As you can see, a `duration` and `arrivalRate` can be defined.
+This way we tell Artillery the amount of load we want to put in the endpoint.
+In this case, it is 20 RPS for 20 seconds, which amounts to 400 requests in total.
+You can also use headers with each request, this is where a "Bearer" token would be used.
+
+To make artillery run this file, simply use the command `$ artillery run my-scenario.yaml`.
+Artillery will then run the scenario and show partial results every 10 seconds.
+The output looks like this:
+
+```
+Report @ 15:43:29(+0000) 2020-11-20
+Elapsed time: 20 seconds
+  Scenarios launched:  0
+  Scenarios completed: 5
+  Requests completed:  5
+  RPS sent: NaN
+  Request latency:
+    min: 195.5
+    max: 219.2
+    median: 202.7
+    p95: 219.2
+    p99: 219.2
+  Codes:
+    200: 5
+
+All virtual users finished
+Summary report @ 15:43:29(+0000) 2020-11-20
+  Scenarios launched:  400
+  Scenarios completed: 400
+  Requests completed:  400
+  RPS sent: 19.68
+  Request latency:
+    min: 192.7
+    max: 586.1
+    median: 201.7
+    p95: 420.3
+    p99: 474.4
+  Scenario counts:
+    0: 400 (100%)
+  Codes:
+    200: 400
+```
+
+Artillery can also:
+* Read values from files and use them in the payloads
+* Use values from the response of previous requests (to chain requests, for example)
+* Define scenarios where RPS increase with time
+* Reporting
+
+For more information you can look in the [documentation](https://artillery.io/docs/guides/overview/welcome.html){:target="_blank"}, it is very easy to understand and also very useful, especially the "guides" section.
+
+## Conclusion
+
+There are many other tools out there: locust, hey, gatling… some even have test scenarios written in code (scala, python…).
```

Comment:
"... Locust, Hey, Gatling… some even have test scenarios written in code (Scala, Python…).